gnocchi/rest/aggregates/api.py

```diff
@@ -0,0 +1,209 @@
+# -*- encoding: utf-8 -*-
+#
+# Copyright © 2016-2017 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import pecan
+from pecan import rest
+import pyparsing
+import six
+import voluptuous
+
+from gnocchi.rest.aggregates import operations as agg_operations
+from gnocchi.rest.aggregates import processor
+from gnocchi.rest import api
+from gnocchi import storage
+from gnocchi import utils
+
+
+def validate_qs(start, stop, granularity, needed_overlap, fill):
+    try:
+        needed_overlap = float(needed_overlap)
+    except ValueError:
+        api.abort(400, 'needed_overlap must be a number')
+    if needed_overlap != 100.0 and start is None and stop is None:
+        api.abort(400, 'start and/or stop must be provided if specifying '
+                  'needed_overlap')
+
+    if start is not None:
+        try:
+            start = utils.to_timestamp(start)
+        except Exception:
+            api.abort(400, "Invalid value for start")
+
+    if stop is not None:
+        try:
+            stop = utils.to_timestamp(stop)
+        except Exception:
+            api.abort(400, "Invalid value for stop")
+
+    if granularity is not None:
+        try:
+            granularity = utils.to_timespan(granularity)
+        except ValueError as e:
+            api.abort(400, e)
```

Comment:
ditto

…and i have the feeling this is not tested because `e` does not have a `jsonify` method so it would fail with a 500.

Reply:
Fixed